### DISCUSSION TITLE
Reports.get_reports() error NextToken cannot be specified with other input parameters when using @load_all_pages

### DISCUSSION THREAD

**Discussion by catinthecode** [saleweaver/python-amazon-sp-api#1022] [Q&A]
`@load_all_pages` is passing all other parameters into `Reports.get_reports`, including `reportTypes`, `processingStatuses`, which is again passed into following queries when it is querying as type `NextToken`, and thus the server responds with this error:

```
sp_api.base.exceptions.SellingApiBadRequestException: [{'code': 'InvalidInput', 'message': 'NextToken cannot be specified with other input parameters', 'details': 'createdUntil;createdSince;processingStatuses;reportTypes;'}]
```

Has anybody seen this? Seems like for some end points, `@load_all_pages` should only pass `NextToken` parameter, instead of together with other parameters (e.g., reportTypes, processingStatuses,createdSince, createdUntil) from the first query, into the following queries of `NextToken` type? And the Amazon end point is returning the above error for these extra parameters.

Thanks

**Comment by FCS-BradyKirby** [comment]
.get_reports(marketplaceIds=None, nextToken=next_token)

The marketplaceIds parameter is getting inferred by default from a prior setting.  Specifying marketplaceIds=None will cause the underlying code to skip the value.

**Comment by catinthecode** [comment]
Logged issue at https://github.com/saleweaver/python-amazon-sp-api/issues/1028.
